docs/src/contributing/development-environment.md

```diff
@@ -1,280 +1,278 @@
-- Start the platform (`deisctl start platform`)
+If your development cluster runs on a cloud provider such as DigitalOcean, a local registry such as the one above will not be accessible to your Kubernetes nodes.  In such cases, a public registry such as [DockerHub][dh] or [quay.io][quay] will suffice.
 
-Conveniently, we can accomplish all three in one step:
+To use DockerHub for this purpose, for instance:
 
-    $ make deploy
+```
+$ export DEIS_REGISTRY=""
+$ export IMAGE_PREFIX=<your DockerHub username>
+```
 
-Running `deisctl list` should display all of the services that your Deis cluster is currently running.
+To use quay.io:
 
-You can now use your Deis cluster in all of the usual ways.
+```
+$ export DEIS_REGISTRY=quay.io/
+$ export IMAGE_PREFIX=<your quay.io username>
+```
 
-At this point, you are running Deis from the code in your Git clone. But since rebuilding like this is time consuming, Deis has a simplified developer workflow more suited to daily development.
+Note the importance of the trailing slash.
 
-## Development Workflow
+### Dev / Deployment Workflow
 
-Deis includes `Makefile` targets designed to simplify the development workflow.
+With a functioning Kubernetes cluster and the officially released Deis components installed onto it, deployment and further testing of any Deis component you have made changes to is facilitated by replacing the officially released component with a custom built image that contains your changes.  Most Deis components include Makefiles with targets specifically intended to facilitate this workflow with minimal friction.
 
-This workflow is typically:
+In the general case, this workflow looks like this:
 
 1. Update source code and commit your changes using `git`
-2. Use `make -C <component> build` to build a new Docker image
-3. Use `make -C <component> dev-release` to push a snapshot release
-4. Use `make -C <component> restart` to restart the component
+2. Use `make build` to build a new Docker image
+3. Use `make dev-release` to generate Kubernetes manifest(s)
+4. Use `make deploy` to restart the component using the updated manifest
 
-This can be shortened to a one-liner using the `deploy` target:
+This can be shortened to a one-liner using just the `deploy` target:
 
-```console
-$ make -C controller deploy
 ```
+$ make deploy
+```
+
+## Useful Commands
 
-You can also use the same tasks on the root `Makefile` to operate on all components at once.  For example, `make deploy` will build, dev-release, and restart all components on the cluster.
+Once your customized Deis component has been deployed, here are some helpful commands that will allow you to inspect your cluster and to troubleshoot, if necessary:
 
-!!! important
-    In order to cut a dev-release, you must commit changes using `git` to increment the SHA used when tagging Docker images.
+### See All Deis Pods
 
-## Test Your Changes
+```
+$ kubectl get pods --namespace=deis
```

Comment:
It's a clever trick and I think we're all using it, but I'm inclined to say we should stick to what's required for development and not embellish too much.